plugins/modules/cloud/opennebula/one_vm.py

```diff
@@ -933,7 +933,11 @@ def create_attributes_str(attributes_dict, labels_list):
     if labels_list:
         attributes_str += 'LABELS="' + ','.join('{label}'.format(label=label) for label in labels_list) + '"\n'
     if attributes_dict:
-        attributes_str += '\n'.join('{key}="{val}"'.format(key=key.upper(), val=val) for key, val in attributes_dict.items()) + '\n'
+        for key, val in attributes_dict.items():
+            if val.lstrip()[0] == "[":
```

Comment:
This fails with an `IndexError: string index out of range` when the `NAME` attribute is not explicitly set (it will be set to an empty string inside the `create_count_of_vms` function).  better would be:

```suggestion
            if len(val) > 0 and val.lstrip()[0] == "[":
```

Reply:
That one can still fail if `val == "  "`. Something that won't (assuming that `val` is always a string) is
```suggestion
            if val.lstrip().startswith("["):
```

Reply:
ah true, nice catch :)

just checked, works fine with `.startswith` as well

Reply:
Hmm, is `val` actually always a string? Shouldn't this only trigger if it is actually a list and not a string?

Also, if this is a string starting with `[`, this change is still a breaking change since so far. Generally, this is not a bugfix, but a new feature, since the module didn't claim to support lists.

Reply:
the module itself at least tries to turn everything attribute it sets on its own into a string, see e.g. https://github.com/ansible-collections/community.general/blob/ab0cd83bb1201c776f67462ff4b5a5d1af677651/plugins/modules/cloud/opennebula/one_vm.py#L1525-L1530

but it doesn't care about what is passed in the `attributes` field from the outside, so `val` could in theory be anything, including a number... which would now break here again (`AttributeError: 'int' object has no attribute 'lstrip'`) :/
It makes sense to somehow enforce the values to be strings then.